src/main.ts

```diff
@@ -32,9 +39,18 @@ if (process.env.NO_CRAWL !== "true") {
       const title = await page.title();
       log.info(`Crawling ${request.loadedUrl}...`);
 
-      await page.waitForSelector(config.selector, {
-        timeout: config.waitForSelectorTimeout ?? 1000,
-      });
+      try {
+        await page.waitForSelector(config.selector, {
+          timeout: config.waitForSelectorTimeout ?? 1000,
+        });
+      } catch (e) {
+        // If the selector is not found, let the user know
+        log.warning(`Selector "${config.selector}" not found on ${request.loadedUrl}, Falling back to "body"`);
+        // using body as a fallback
+        await page.waitForSelector("body", {
+          timeout: config.waitForSelectorTimeout ?? 1000,
+        });
```

Comment:
**Suggestion:** Refactor the code to avoid code duplication when waiting for selectors. You can create a helper function that accepts a selector and timeout as parameters and use it in both cases.
```suggestion
      async function waitForSelectorOrFallback(page: Page, selector: string, fallbackSelector: string, timeout: number) {
        try {
          await page.waitForSelector(selector, { timeout });
        } catch (e) {
          log.warning(`Selector "${selector}" not found, Falling back to "${fallbackSelector}"`);
          await page.waitForSelector(fallbackSelector, { timeout });
        }
      }

      await waitForSelectorOrFallback(page, config.selector, "body", config.waitForSelectorTimeout ?? 1000);
```

Reply:
thanks for the suggestions, i just added those!